src/engine/structures/lock.rs

```diff
@@ -370,6 +370,7 @@ where
     /// Returns true if:
     /// * The waiting task does not conflict with any already woken tasks.
     /// * The waiting task does not conflict with any locks currently held.
+    #[allow(clippy::get_first)]
```

Comment:
Instead of the allow, use ```front()``` for ```get(0)```.

Reply:
OK; now I see it.  It turns out that clippy did recommend `front()`, but it was in a large list of `first()` recommendations.  I'll revert the clippy allow, let this change run through CI, and then rebase to remove the unnecessary allow commit if all tests pass.